posthog/models/organization.py

```diff
@@ -45,6 +45,10 @@ class Organization(UUIDModel):
     name: models.CharField = models.CharField(max_length=64)
     created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True)
     updated_at: models.DateTimeField = models.DateTimeField(auto_now=True)
+    setup_last_completed_section: models.IntegerField = models.IntegerField(
```

Comment:
A better default would be null=True.

So this denormalizes the data model quite a bit and creates the inconsistency I was worried about in the frontend where some things use one data source, others another.

We know whether user has finished step 1 is derivable from the same flags as in the frontend (ingested event + completed snippet onboarding).

For step 2, I think a single boolean field is more flexible long-term.

Reply:
1. Sure, I'll update to `None` as default.
2. I'm not sure I follow. We would now use `setup_last_completed_section` as the only source of truth to determine whether to show the demo warning & setup page, and which **section** to enable. The rest of the variables would be used only if the user is on _Section 1_ to determine on which step (subdivision) of Section 1 the user is in.
3. I don't think that'll be a problem. For now the migration defaults to `setup_last_completed_section = 3` (i.e. marked as completed). We only enable the process for users if the feature flag is active (see `posthog/api/organization.py`).

Reply:
> I'm not sure I follow. We would now use setup_last_completed_section as the only source of truth to determine whether to show the demo warning & setup page, and which section to enable. The rest of the variables would be used only if the user is on Section 1 to determine on which step (subdivision) of Section 1 the user is in.

There's several problems here:
1. Confusing semantics - I'm not sure what step=0 or step=1 mean based on just this code. I'm assuming step=1 means step 1 has been completed below but this is not obvious.
1. Information duplication - values 0 and 1 can be inferred from two other values (any_team_completed_onboarding, ...). If we duplicate information, we introduce risk of e.g. user deleting a team and then us showing the warning (due to steps logic) but the actual db information being off. It also complicates debugging + extending this flow going forward. 

A flag for "has clicked continue on step 2" is simpler. Note I don't advocate sending that flag to fe but instead calculating the step in the serializer.

> I don't think that'll be a problem. For now the migration defaults to setup_last_completed_section = 3 (i.e. marked as completed). We only enable the process for users if the feature flag is active (see posthog/api/organization.py).

This introduces a problem: migration will run before the flag will be enabled. So anyone who signs up between the migration running and the flag being turned on will have broken state.

This is why we should set it explicitly as None in organization creation API endpoint if not enabling new onboarding.

Reply:
Forgot 3. - It makes every change to setup page involve tricky migations - something best avoided.

Reply:
For posterity, as agreed on sync call

- We'll do the simple boolean for section 2 (as when we introduce section 3 we'll have to store special information anyways).
- We'll treat by default (on migration level) the steps as completed, and on the serializer we'll check if the feature flag is enabled to send the user through the onboarding flow.